### PR TITLE
Matches single without selector

### PR DESCRIPTION
**Pull request by wojtek-t**
This is reducing cpu-usage of apiserver by ~5% in 5000-node clusters.

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go [hunk 1]
```diff
@@ -65,16 +65,42 @@ func (s *SelectionPredicate) MatchesLabelsAndFields(l labels.Set, f fields.Set)
 	return matched
 }
 
+// TODO: should be namespace.name
+const matchesSingleField = "metadata.name"
```
Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go]: Why should this be `namespace.name`?

**Comment by smarterclayton** [comment]
One question

File: staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go (hunk 1 above)
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go]: I copied it from the other place. I'm not sure why it was there.

Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go]: It's wrong - "metadata.name" *is* the correct field (`{"metadata":{"name": ...}}`).  Strip it and this LGTM

Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go]: Done.

**Comment by wojtek-t** [comment]
Applying label based on @smarterclayton comment above - thanks!

**Comment by wojtek-t** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by smarterclayton** [comment]
I may have regressed / altered the behavior of this with the initializers pull. Most calls have to use a predicate for filtering out uninitialized resources, so I'm unsure whether this still gets the benefit you expected.

**Comment by wojtek-t** [comment]
> I may have regressed / altered the behavior of this with the initializers pull. Most calls have to use a predicate for filtering out uninitialized resources, so I'm unsure whether this still gets the benefit you expected.

@smarterclayton - thanks for the update. And no worries, the gain will be smaller, but there should still be some. And it wasn't critical to anything.

**Pull request merged**